File: benchmarks/python/matrix.py
#!/usr/bin/env python3

# Matrix operations benchmark in Python

def multiply_matrices(a, b, size):
    result = [[0 for _ in range(size)] for _ in range(size)]
    for i in range(size):
        for j in range(size):
            for k in range(size):
                result[i][j] += a[i][k] * b[k][j]
    return result

def create_matrix(size):
    matrix = []
    for i in range(size):
        row = []
        for j in range(size):
            row.append(i * size + j + 1)
        matrix.append(row)
    return matrix

# Create 20x20 matrices
size = 20
matrix_a = create_matrix(size)
matrix_b = create_matrix(size)

# Multiply matrices
result = multiply_matrices(matrix_a, matrix_b, size)
print(f"Matrix {len(result)}x{len(result[0])} = {result[0][0]}")

File: benchmarks/javascript/matrix.js
#!/usr/bin/env bun

// Matrix operations benchmark in JavaScript

function multiplyMatrices(a, b, size) {
    const result = Array(size).fill().map(() => Array(size).fill(0));
    for (let i = 0; i < size; i++) {
        for (let j = 0; j < size; j++) {
            for (let k = 0; k < size; k++) {
                result[i][j] += a[i][k] * b[k][j];
            }
        }
    }
    return result;
}

function createMatrix(size) {
    const matrix = [];
    for (let i = 0; i < size; i++) {
        const row = [];
        for (let j = 0; j < size; j++) {
            row.push(i * size + j + 1);
        }
        matrix.push(row);
    }
    return matrix;
}

// Create 20x20 matrices
const size = 20;
const matrixA = createMatrix(size);
const matrixB = createMatrix(size);

// Multiply matrices
const result = multiplyMatrices(matrixA, matrixB, size);
console.log(`Matrix ${result.length}x${result[0].length} = ${result[0][0]}`);